gui-tk/godley.tcl

```diff
@@ -156,6 +156,90 @@ proc mouseDown {id x y X Y} {
     }
 }
 
+# warn user when a stock variable column is going to be moved to a different asset class on pressing a column button widget. For ticket 1072.
```

Comment:
There is too much "business logic" wrapped up in TCL code for my liking. Given we're planning on replacing TCL code with Javascript code, it would be better to find a way of managing this process with as much business logic done in C++ as possible.

I can think of two approaches: 

1) a C++ method that detects whether the column move changes asset classes and if so returns back to TCL to post the dialog. If not then the column change method is called directly. In the case of the TCL dialog approving the column change, then call into the same C++ method from TCL to do it.

2) The alternative method is for C++ to call back into TCL to post the dialog. In order to do this, but not infect the main C++ code with too much TCL stuff is to use a virtual function override, much like the clipboard copy/paste operations.

Reply:
I understand and should have had the Javascript code in mind when I did this... 

For the second option, would the code that I need to add to minskyTCL.cc be platform independent? In the case of the clipboard, I understand that this is not the case because different platforms handle clipboard differently, but I'm not sure in this case.

Reply:
I took the first approach in the end. I tried to combine both swapAssetClass and moveAssetClass methods into one but hit a snag in TCL when I tried to use the variable buttonPressed (with values 0 or 1) to switch between (mouseUp and mouseDown). Obviously, mouseDown or buttonPressed=1 always gets activated first, so I spilt the method back into the two I originally had. 

As an aside, I tried to fix the right arrow/tab (navigateRight()) and left arrow/back tab traversal (navigateLeft()) methods to traverse all editable cells in a closed loop, but only succeeded in the first case. Back tab will only traverse cells from the last one up to the first in the initial conditions row and the left arrow will go all the way to the first editable cell but then stay in the first row upon continued navigateLeft(), i.e., it does not start from the last cell in the table.